### ISSUE TITLE
Flake: ClientLbEnd2endTest.RoundRobinReresolve GRPC_POLL_STRATEGY=poll 

### ISSUE THREAD

**Issue by ncteisen** [grpc/grpc#13269]
```
Failure
D1104 03:23:37.509646818   22436 test_config.c:388]          test slowdown factor: sanitizer=1, fixture=1, poller=1, total=1
D1104 03:23:37.509878799   22436 ev_posix.cc:135]            Using polling engine: poll
D1104 03:23:37.509905517   22436 dns_resolver.cc:307]        Using native dns resolver
Note: Google Test filter = ClientLbEnd2endTest.RoundRobinReresolve
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from ClientLbEnd2endTest
[ RUN      ] ClientLbEnd2endTest.RoundRobinReresolve
I1104 03:23:37.517991527   22436 client_lb_end2end_test.cc:192] starting server on port 12813
I1104 03:23:37.518149116   22496 server_builder.cc:269]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I1104 03:23:37.519769694   22436 client_lb_end2end_test.cc:200] server startup complete
I1104 03:23:37.519789426   22436 client_lb_end2end_test.cc:192] starting server on port 32167
I1104 03:23:37.520519344   22500 server_builder.cc:269]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I1104 03:23:37.522149529   22436 client_lb_end2end_test.cc:200] server startup complete
I1104 03:23:37.522162539   22436 client_lb_end2end_test.cc:192] starting server on port 32399
I1104 03:23:37.522498180   22502 server_builder.cc:269]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I1104 03:23:37.524059367   22436 client_lb_end2end_test.cc:200] server startup complete
E1104 03:23:37.542018359   22483 round_robin.cc:280]         assertion failed: !p->shutdown



*******************************
Caught signal SIGABRT
bins/opt/client_lb_end2end_test[0x57ccde]
/lib/x86_64-linux-gnu/libpthread.so.0(+0xf890)[0x7ffa79fbe890]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x37)[0x7ffa791f6067]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x148)[0x7ffa791f7448]
bins/opt/client_lb_end2end_test[0x4b535b]
bins/opt/client_lb_end2end_test[0x484260]
bins/opt/client_lb_end2end_test[0x4855c2]
bins/opt/client_lb_end2end_test[0x43ab6f]
bins/opt/client_lb_end2end_test[0x43d048]
bins/opt/client_lb_end2end_test[0x4544fd]
bins/opt/client_lb_end2end_test[0x4acc1e]
bins/opt/client_lb_end2end_test[0x43d024]
bins/opt/client_lb_end2end_test[0x444cf1]
bins/opt/client_lb_end2end_test[0x46f0b7]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x8064)[0x7ffa79fb7064]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7ffa792a962d]
```

**Comment by dgquintas** [comment]
This will be fixed by https://github.com/grpc/grpc/pull/12427, which I'll merge today.

**Comment by dgquintas** [comment]
Fixed by #12427
